vault_service/views/utils.py

```diff
@@ -85,7 +91,425 @@ def check_request(request):
     if headers['Authorization']:
         new_headers['X-Forwarded-Authorization'] = headers['Authorization']
     new_headers['Authorization'] = 'Bearer:' + current_app.config['VAULT_OAUTH_CLIENT_TOKEN']
-    new_headers['X-Adsws-Uid'] = headers.get('X-Adsws-Uid', '0') # User ID
+    new_headers['X-Adsws-Uid'] = headers.get('X-Adsws-Uid', str(current_app.config['BOOTSTRAP_USER_ID'])) # User ID
 
     return (payload, new_headers)
 
+
+def upsert_myads(classic_setups, user_id):
+    # check to see if user has a myADS setup already
+    with current_app.session_scope() as session:
+        try:
+            q = session.query(User).filter_by(id=user_id).one()
+        except ormexc.NoResultFound:
+            u = User(id=user_id)
+            try:
+                session.add(u)
+                session.commit()
+            except exc.IntegrityError as e:
+                session.rollback()
+                return json.dumps({'msg': 'User does not exist, so new myADS setup was not saved, error: {0}'.
+                                  format(e)}), 500
+
+    new_setups = []
+    existing_setups = []
+    # painstakingly step through all the keys in the Classic setup
+    # u'id' --> user ID - int
+    # u'email' --> user email
+    # u'firstname' --> user first name (for citations)
+    # u'lastname' --> user last name (for citations)
+    # u'daily_t1,' --> keywords 1 (daily arxiv)
+    # u'groups' --> arxiv classes (daily) - list
+    # u'phy_t1,' --> keywords 1 (physics)
+    # u'phy_t2,' --> keywords 2 (physics)
+    # u'phy_aut,' --> authors (physics)
+    # u'pre_t1,' --> keywords 1 (weekly arxiv)
+    # u'pre_t2,' --> keywords 2 (weekly arxiv)
+    # u'pre_aut,' --> authors (weekly arxiv)
+    # u'ast_t1,' --> keywords 1 (astronomy)
+    # u'ast_t2,' --> keywords 2 (astronomy)
+    # u'ast_aut,' --> authors (astronomy)
+
+    if len(classic_setups.get('lastname', '')) > 0:
```

Comment:
This function could be splitted at least creating a function for each of the `if` that are at this level.